src/test/kotlin/com/gamingbacklog/api/gamingbacklogapi/controllers/LibraryControllerTests.kt

```diff
@@ -1,153 +1,175 @@
     }
+    /** Tests returning 204 for getting all libraries */
+    @Test
+    fun shouldReturnNoValues() {
+      given(libraryService.getAll()).willReturn(emptyList())
+      requestBuilder.runGetRequest(endpoint)
+        .andExpect(status().isOk)
+        .andExpect(jsonPath("$", equalTo(emptyList<Library>())))
+    }
+  }
 
-    @Nested
-    @DisplayName("Tests for getLibraries")
-    inner class GetLibraries {
-        /** Tests returning a 200 response for returning all libraries */
-        @Test
-        fun shouldReturnAllLibraries() {
-            val library1 = Library("id1", "Backlog", ArrayList())
-            val library2 = Library("id2", "Owned Games", ArrayList())
-            library2.games.add("gameId1")
-            library2.games.add("gameId2")
-            val libraries = ArrayList<Library>()
-            libraries.add(library1)
-            libraries.add(library2)
-            given(libraryService.getAll()).willReturn(libraries)
-            requestBuilder.runGetRequest(endpoint)
-                    .andExpect(status().isOk)
-                    .andExpect(jsonPath("$[0].name", equalTo("Backlog")))
-                    .andExpect(jsonPath("$[0].games", equalToObject(ArrayList<String>())))
-                    .andExpect(jsonPath("$[0].id", equalTo("id1")))
-                    .andExpect(jsonPath("$[1].name", equalTo("Owned Games")))
-                    .andExpect(jsonPath("$[1].games[0]", equalTo("gameId1")))
-                    .andExpect(jsonPath("$[1].games[1]", equalTo("gameId2")))
-                    .andExpect(jsonPath("$[1].id", equalTo("id2")))
-        }
-        /** Tests returning 204 for getting all libraries */
-        @Test
-        fun shouldReturnNoValues() {
-            given(libraryService.getAll()).willReturn(emptyList())
-            requestBuilder.runGetRequest(endpoint)
-                    .andExpect(status().isOk)
-                    .andExpect(jsonPath("$", equalTo(emptyList<Library>())))
-        }
+  @Nested
+  @DisplayName("Tests for getSingleLibrary")
+  inner class GetSingleLibrary {
+    // returns 200
+    @Test
+    fun shouldReturnLibrarySuccessfully() {
+      val library = Library(id1, "Owned Games", ArrayList())
+      library.games.add("gameId1")
+      library.games.add("gameId2")
+      given(libraryService.getSingle(any())).willReturn(library)
+      endpoint += "$id1/"
+      requestBuilder.runGetRequest(endpoint)
+        .andExpect(status().isOk)
+        .andExpect(jsonPath("$.name", equalTo("Owned Games")))
+        .andExpect(jsonPath("$.games[0]", equalTo("gameId1")))
+        .andExpect(jsonPath("$.games[1]", equalTo("gameId2")))
+        .andExpect(jsonPath("$.id", equalTo(id1)))
     }
+    // returns 404
+    @Test
+    fun shouldReturnNoLibrary() {
+      given(libraryService.getSingle(any())).willReturn(null)
+      endpoint += "$id1/"
+      requestBuilder.runGetRequest(endpoint)
+        .andExpect(status().isOk)
```

Comment:
Isn't 404 usually an error status? How would this return ok?

Reply:
Came to this realization too while working on my next story. I've started making the changes to actually support 404 there in the code already so I'll have this addressed in my PR for https://gaming-backlog.atlassian.net/browse/GB-46 (hate to bloat too much but it's already worked in lol)